shotover/src/codec/cassandra.rs

```diff
@@ -465,6 +473,15 @@ fn set_startup_state(
         );
     }
 
+    if direction == Direction::Source {
+        match version {
+            Version::V3 => increment_counter!("client_protocol_version", "version" => "v3"),
```

Comment:
Is it possible to use a metric handle here for better performance?

Reply:
Good point, ending up finding some optimisations where `CassandraCodecBuilder` was being created unnecessarily.